File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    while(1) {
        vector<int> v;
        cin >> n;
        if (n == -1) {
            break;
        }
        
        int sum = 0;
        for (int i = 1; i <= n / 2; i++) {
            if (n % i == 0) {
                v.push_back(i);
                sum += i;
            }
        }
        
        if (n == sum) {
            cout << n << " = ";
            for (int i = 0; i < v.size() - 1; i++) {
                cout << v[i] << " + ";
            }
            cout << v[v.size() - 1] << '\n';
        } else {
            cout << n << " is NOT perfect." << '\n';
        }
    }
    return 0;
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while(True):
    N = int(input())
    
    if N == -1:
        break
        
    list1 = []
    sum = 0
    for i in range(1, N):
        if N % i == 0:
            list1.append(i)
            sum += i
    if sum == N:
        result = " + ".join(str(i) for i in list1)
        print(f"{N} = {result}")
    else:
        print(str(N) + " is NOT perfect.")
    
            
        